src/plugins/template/src/variable_state.hpp

```diff
@@ -7,22 +7,28 @@
 #include "openvino/runtime/itensor.hpp"
 #include "openvino/runtime/ivariable_state.hpp"
 #include "openvino/runtime/so_ptr.hpp"
+#include "openvino/op/util/variable.hpp"
 
 namespace ov {
 namespace template_plugin {
 
 class VariableState : public ov::IVariableState {
 public:
-    VariableState(const std::string& name, const std::shared_ptr<op::util::VariableValue>& variable_value)
-        : ov::IVariableState(name),
+    VariableState(const ov::op::util::VariableInfo& variable_info,
+                  const std::shared_ptr<op::util::VariableValue>& variable_value)
+        : ov::IVariableState(variable_info.variable_id),
+          m_data_shape(variable_info.data_shape),
+          m_data_type(variable_info.data_type),
           m_variable_value(variable_value) {
         m_state = get_tensor_impl(variable_value->get_state());
     }
     void set_state(const ov::SoPtr<ov::ITensor>& state) override {
-        OPENVINO_ASSERT(state->get_shape() == m_state->get_shape(), "Wrong tensor shape.");
-        OPENVINO_ASSERT(state->get_element_type() == m_state->get_element_type(), "Wrong tensor type.");
+        OPENVINO_ASSERT(m_data_shape.compatible(state->get_shape()), "Wrong tensor shape");
```

Comment:
minor: it's better to add ore details in the message,
 e.g.
a new state->get_shape() shape have to be compatible with m_data_shape variable shape.

Reply:
Done